src/main/java/org/snomed/snowstorm/core/data/services/AutoMerger.java

```diff
@@ -0,0 +1,518 @@
+                mergedReferenceSetMember.setActive(true);
+            }
+            mergedReferenceSetMembers.put(changedLangMemberId, mergedReferenceSetMember);
+        }
+
+        // Re-populate Description's LanguageReferenceSetMembers
+        for (Description description : mergedConcept.getDescriptions()) {
+            String descriptionId = description.getDescriptionId();
+            Map<String, ReferenceSetMember> languageReferenceSetMembers = new HashMap<>();
+
+            // Replace existing with newly merged
+            for (ReferenceSetMember referenceSetMember : description.getLangRefsetMembers()) {
+                String memberId = referenceSetMember.getMemberId();
+                if (changedLangMemberIds.contains(memberId)) {
+                    languageReferenceSetMembers.put(memberId, mergedReferenceSetMembers.get(memberId));
+                    mergedReferenceSetMembers.remove(memberId);
+                } else {
+                    languageReferenceSetMembers.put(memberId, referenceSetMember);
+                }
+            }
+
+            // Add newly created
+            for (ReferenceSetMember referenceSetMember : mergedReferenceSetMembers.values()) {
+                String memberId = referenceSetMember.getMemberId();
+                String referencedComponentId = referenceSetMember.getReferencedComponentId();
+
+                // Keep language associated with correct Description
+                if (referencedComponentId.equals(descriptionId)) {
+                    languageReferenceSetMembers.putIfAbsent(memberId, mergedReferenceSetMembers.get(memberId));
+                }
+            }
+
+            description.setLanguageRefsetMembers(languageReferenceSetMembers.values());
+        }
+    }
+
+    // valueDefault is coming from Source and may have newer/different value than valueOld.
+    private String getValueChanged(String valueOld, String valueNew, String valueDefault) {
+        boolean valueHasChanged = !Objects.equals(valueOld, valueNew);
+        if (valueHasChanged) {
+            return valueNew;
+        } else {
+            return valueDefault;
+        }
+    }
+
+    // valueDefault is coming from Source and may have newer/different value than valueOld.
+    private Boolean getValueChanged(Boolean valueOld, Boolean valueNew, Boolean valueDefault) {
+        boolean valueHasChanged = !Objects.equals(valueOld, valueNew);
+        if (valueHasChanged) {
+            return valueNew;
+        } else {
+            return valueDefault;
+        }
+    }
+
+    // valueDefault is coming from Source and may have newer/different value than valueOld.
+    private Integer getValueChanged(Integer valueOld, Integer valueNew, Integer valueDefault) {
+        boolean valueHasChanged = !Objects.equals(valueOld, valueNew);
+        if (valueHasChanged) {
+            return valueNew;
+        } else {
+            return valueDefault;
+        }
+    }
```

Comment:
These are util methods for working out what the author has changed. If the author hasn't changed the field, then use the field from the Source, i.e. valueDefault.